backend/server/controllers/user.js

```diff
@@ -29,10 +31,35 @@ export const getUsers = async (req, res) => {
 
 export const getUsersById = async (req, res) => {
   try {
-    const { params } = req;
-    const user = await models.users.findByPk(params.id);
+    // const { params } = req;
+    // const user = await models.users.findByPk(params.id);
+    /** Instead of the above commented code, should use authToken
+     * provided in the request header
+     */
+    const authToken = req.get('authToken');
+    const user = await Authentication.extractUser(authToken);
     res.status(200).send(user);
   } catch (error) {
-    res.status(500).send(error);
+    res.status(500).send({"Error message": error.toString()});
   }
 };
```

Comment:
This would be the endpoint used to view a user's profile so the logic shouldn't be changed. We just need to extract the authToken from the request header (the request is saved in the `req` variable) and then check the authToken is valid to make sure the user sending the request is authorized. But the existing logic should be left as it is